pbrd/pbr_vty.c

```diff
@@ -153,325 +128,949 @@ DEFUN_NOSH(no_pbr_map, no_pbr_map_cmd, "no pbr-map PBRMAP [seq (1-700)]",
+		vty_out(vty, "Unable to convert %s to proto id\n",
+			(ip_proto ? ip_proto : "(null)"));
 		return CMD_WARNING_CONFIG_FAILED;
 	}
 
-	pbrms->family = prefix->family;
+	pbrms->ip_proto = p->p_proto;
+	SET_FLAG(pbrms->filter_bm, PBR_FILTER_IP_PROTOCOL);
 
-	if (!no) {
-		if (pbrms->src) {
-			if (prefix_same(pbrms->src, prefix))
-				return CMD_SUCCESS;
-		} else
-			pbrms->src = prefix_new();
+check:
+	pbr_map_check(pbrms, true);
+	return CMD_SUCCESS;
+}
 
-		prefix_copy(pbrms->src, prefix);
-	} else
-		prefix_free(&pbrms->src);
+/* clang-format off */
+DEFPY  (pbr_map_match_src_port,
+	pbr_map_match_src_port_cmd,
+	"[no] match src-port ![(1-65535)$port]",
+	NO_STR
+	"Match the rest of the command\n"
+	"Choose the source port to use\n"
+	"The Source Port\n")
+{
+	/* clang-format on */
+	struct pbr_map_sequence *pbrms = VTY_GET_CONTEXT(pbr_map_sequence);
+
+	if (!pbrms)
+		return CMD_WARNING_CONFIG_FAILED;
+
+	if (no) {
+		if (!CHECK_FLAG(pbrms->filter_bm, PBR_FILTER_SRC_PORT))
+			return CMD_SUCCESS;
+		UNSET_FLAG(pbrms->filter_bm, PBR_FILTER_SRC_PORT);
+		goto check;
+	}
 
+	if (CHECK_FLAG(pbrms->filter_bm, PBR_FILTER_SRC_PORT) &&
+	    (pbrms->src_prt == port)) {
+		return CMD_SUCCESS;
+	}
+	pbrms->src_prt = port;
+	SET_FLAG(pbrms->filter_bm, PBR_FILTER_SRC_PORT);
+
+check:
 	pbr_map_check(pbrms, true);
+	return CMD_SUCCESS;
+}
 
+/* clang-format off */
+DEFPY  (pbr_map_match_dst_port,
+	pbr_map_match_dst_port_cmd,
+	"[no] match dst-port ![(1-65535)$port]",
+	NO_STR
+	"Match the rest of the command\n"
+	"Choose the destination port to use\n"
+	"The Destination Port\n")
+{
+	/* clang-format on */
+	struct pbr_map_sequence *pbrms = VTY_GET_CONTEXT(pbr_map_sequence);
+
+	if (!pbrms)
+		return CMD_WARNING_CONFIG_FAILED;
+
+	if (no) {
+		if (!CHECK_FLAG(pbrms->filter_bm, PBR_FILTER_DST_PORT))
+			return CMD_SUCCESS;
+		UNSET_FLAG(pbrms->filter_bm, PBR_FILTER_DST_PORT);
+		goto check;
+	}
+
+	if (CHECK_FLAG(pbrms->filter_bm, PBR_FILTER_DST_PORT) &&
+	    (pbrms->dst_prt == port)) {
+		return CMD_SUCCESS;
+	}
+	pbrms->dst_prt = port;
+	SET_FLAG(pbrms->filter_bm, PBR_FILTER_DST_PORT);
+
+check:
+	pbr_map_check(pbrms, true);
 	return CMD_SUCCESS;
 }
 
-DEFPY(pbr_map_match_dst, pbr_map_match_dst_cmd,
-	"[no] match dst-ip <A.B.C.D/M|X:X::X:X/M>$prefix",
+/* clang-format off */
+DEFPY  (pbr_map_match_dscp,
+	pbr_map_match_dscp_cmd,
+	"[no] match dscp ![DSCP$dscp]",
 	NO_STR
 	"Match the rest of the command\n"
-	"Choose the dst ip or ipv6 prefix to use\n"
+	"Match based on IP DSCP field\n"
+	"DSCP value (below 64) or standard codepoint name\n")
+{
+	/* clang-format on */
+	struct pbr_map_sequence *pbrms = VTY_GET_CONTEXT(pbr_map_sequence);
+
+	if (!pbrms)
+		return CMD_WARNING_CONFIG_FAILED;
+
+	if (no) {
+		if (!CHECK_FLAG(pbrms->filter_bm, PBR_FILTER_DSCP))
+			return CMD_SUCCESS;
+		UNSET_FLAG(pbrms->filter_bm, PBR_FILTER_DSCP);
+		pbrms->dsfield &= ~PBR_DSFIELD_DSCP;
+		goto check;
+	}
+
+	long lDscp;
```

Comment:
Can we avoid mixing variable naming schemes (camelCase / snake_case)?

Reply:
Ok. Changed.